### PR TITLE
Add custom detection rule to detect Bearer token leaks.

### PR DESCRIPTION
**Pull request by akiioto**
**Description**

Changes proposed in this pull request:

- Add custom detection rule to detect Bearer token leaks.

**Related issue(s)**

### CODE REVIEW

**Comment by akiioto** [comment]
pre-test-infra-validate-dockerfiles

File: development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go [hunk 1]
```diff
@@ -63,14 +63,17 @@ func main() {
 		mainLogger.LogCritical("failed to create storageClient: %s", err)
 	}
 
-	defer storageClient.Close()
-
 	// Setup configuration for gitleaks
-	viper.SetConfigType("toml")
-	if err := viper.ReadConfig(strings.NewReader(config.DefaultConfig)); err != nil {
-		mainLogger.LogError("failed read default viper config for leaks scanner, error: %s", err)
+	viper.SetConfigFile("app/development/gcp/cloud-run/scan-logs-for-secrets/gitleaks.toml")
```
Comment (KacperMalachowski) [inline development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go:67]: This file path can be simplified with a fixed Dockerfile

**Review by KacperMalachowski — CHANGES_REQUESTED** [review]

File: development/gcp/cloud-run/scan-logs-for-secrets/Dockerfile [hunk 2]
```diff
@@ -8,6 +8,8 @@ RUN  CGO_ENABLED=0 go build -o /scanlogsforsecrets -ldflags="-s -w" ./developmen
 FROM alpine:3.17.1
 
 COPY --from=builder /scanlogsforsecrets /scanlogsforsecrets
+COPY --from=builder /go/src/github.com/kyma-project/test-infra/ /app/
+
```
[the remark this answers is not in the document]
Comment (akiioto) [reply development/gcp/cloud-run/scan-logs-for-secrets/Dockerfile:12]: Good, I will rework this.

File: development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go (hunk 1 above)
Comment (akiioto) [reply development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go:67]: Oh, I see. The same code is being executed twice. I will correct this.

File: development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go [hunk 3]
```diff
@@ -63,14 +63,17 @@ func main() {
...
+	viper.ReadInConfig()
+	if err := viper.ReadInConfig(); err != nil {
```
[the remark this answers is not in the document]
Comment (akiioto) [reply development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go:69]: Oh, I see. The same code is being executed twice. I will correct this.

File: development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go [hunk 4]
```diff
@@ -63,14 +63,16 @@ func main() {
 		mainLogger.LogCritical("failed to create storageClient: %s", err)
 	}
 
-	defer storageClient.Close()
-
 	// Setup configuration for gitleaks
-	viper.SetConfigType("toml")
-	if err := viper.ReadConfig(strings.NewReader(config.DefaultConfig)); err != nil {
-		mainLogger.LogError("failed read default viper config for leaks scanner, error: %s", err)
+	viper.SetConfigFile("app/gitleaks.toml"
```
Comment (halamix2) [inline development/gcp/cloud-run/scan-logs-for-secrets/scan-logs-for-secrets.go:67]:
```suggestion
	viper.SetConfigFile("app/gitleaks.toml")
```

**Review by halamix2 — CHANGES_REQUESTED** [review]

**Review by KacperMalachowski — APPROVED** [review]
LGTM

**Review by halamix2 — APPROVED** [review]

**Pull request merged**